src/com/esotericsoftware/kryo/serializers/DefaultSerializers.java

```diff
@@ -852,6 +853,27 @@ private PriorityQueue createPriorityQueue (Class<? extends Collection> type, int
 		}
 	}
 
+	/** Serializer for {@link ConcurrentHashMap.KeySetView}.
+	 * @author Andreas Bergander */
+	public static class KeySetViewSerializer extends CollectionSerializer<ConcurrentHashMap.KeySetView> {
+		protected void writeHeader (Kryo kryo, Output output, ConcurrentHashMap.KeySetView set) {

```

Comment:
I just realized that this is not ideal. We are writing the keys twice. Once here as part of the map in the header and then again in the main `write` method of the `CollectionSerializer`. It would be enough to write the map because the keyset is only a view. Can you try to implement this without extending `CollectionSerializer` and implement `Serializer` directly? This should make the serialized payload significantly smaller.

Reply:
`writeHeader` would simply become `write` and `create` would become `read`.

Reply:
Thank you for pointing me in the right direction! PR is updated according to your comments.